### ISSUE TITLE
Loading screen while fetching the data

### ISSUE THREAD

**Issue by shubhsk88** [nthopinion/covid19#55]
When the client is fetching the data from API there is no loading screen for that

**Comment by nastikue** [comment]
I've found an interesting article that can be helpful when issue is reopened https://medium.com/@Charles_Stover/react-suspense-with-the-fetch-api-a1b7369b0469

**Comment by aviskarkc10** [comment]
@nastikue Not sure if we should use react suspense right away. It's not production ready yet:
https://reactjs.org/docs/concurrent-mode-suspense.html

BTW have you worked with Suspense and data fetching previously?

**Comment by nastikue** [comment]
@aviskarkc10 I've researched Suspense for the other project and we decided not to use it there, agree that it should be used in production until it is stable.

**Comment by shubhsk88** [comment]
I researched about that it as well Maybe we can do the react-infinite scrolling and fetch the data after when user slide it down in that way we don't have to get all the data at once and we can show some data and when the user slides it down we can fetch our data
Please look up more about this article
https://www.smashingmagazine.com/2020/03/infinite-scroll-lazy-image-loading-react/

**Comment by nastikue** [comment]
@shubhsk88 the article starts with "If you have been looking for an alternative to pagination", so I am guessing why not to make pagination then?

**Comment by aviskarkc10** [comment]
Yeah the best option now is either implementing pagination/infinite scroll. Having this discussion on one of the slack channels and proceeding based on the decision made their should be the best way forward.
